tests/code_size/random_printf_wasm2js.json

```diff
@@ -1,6 +1,6 @@
 {
-  "a.html": 17361,
-  "a.html.gz": 7503,
-  "total": 17361,
-  "total_gz": 7503
+  "a.html": 17365,
+  "a.html.gz": 7507,
+  "total": 17365,
+  "total_gz": 7507
 }
```

Comment:
It seems that this is due to closure being less able to combine declaration in some cases.  For example:

```
-  for (var c = 0, d = a.length - 1; 0 <= d; d--) {
-    var e = a[d];
-    "." === e ? a.splice(d, 1) : ".." === e ? (a.splice(d, 1), c++) : c && (a.splice(d, 1), c--);
+  var c = 0;
+  for (let e = a.length - 1; 0 <= e; e--) {
+    var d = a[e];
+    "." === d ? a.splice(e, 1) : ".." === d ? (a.splice(e, 1), c++) : c && (a.splice(e, 1), c--);
```

The the new version `c` was able to be declared in the for statement itself, but in the version the `let` and `var`s have to be declared separately.

I reverted a few of the cases where we were seeing this behaviour, but the differences are very minimal and within the normal tolerances.

Reply:
That makes me worry, though, how will we know when to use `let` in the future? Is the idea to not mix them in the same function? If that's right then lgtm.

Reply:
Do you think we need to have a rule for this?    There are a lot of things we could do that help closure micro optimize better.   For example, we could say that variable names should be re-used whereever possible within a function, to minimize the total number of variable.. but we don't currently have rules for that.

I guess if we did have a rule it would be "don't mix let and var within the same function"... I guess that would ok?

Reply:
I guess I'm worried less about a rule, and more having an idea of what the cost is. And that summary sgtm.